assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js

```diff
@@ -83,7 +83,7 @@ function DashboardSummaryWidget() {
 		return getDataErrorComponent( 'adsense', error.message, false, false, false, error );
 	}
 
-	if ( ! today?.totals && ! period?.totals && ! daily?.totals ) {
+	if ( isZeroReport( today ) && isZeroReport( period ) && isZeroReport( daily ) ) {
```

Comment:
I'm curious why `today`, `period`, and/or `daily` would be `undefined` here given `loading` indicates that they're still loading. I guess that `isResolving` selector used to load `loading` might not return `true` on first render here?

If so, I think it's best to change the `if ( loading )` check above to check for `if ( loading || today === undefined || period === undefined || daily === undefined )`.

I think it's better not to have undefined reports (eg ones that are loading) show a zero data state, as that's different to loading and could be confusing.

Reply:
@tofumatt it happened because of the wrong order of selectors call. `isResolving` selectors must be called after `getReport` ones. I have updated it in the component and now everything works correctly.

Reply:
Ah, awesome, good catch.